fluent-react/src/localized.js

```diff
@@ -116,12 +100,36 @@ export default class Localized extends Component {
     }
 
     const msg = mcx.getMessage(id);
-    const args = toArguments(this.props);
-    const { parts, attrs } = l10n.formatCompound(mcx, msg, args);
+    const [args, elems] = toArguments(this.props);
+    const { value, attrs } = l10n.formatCompound(mcx, msg, args);
+
+    if (value === null || !value.includes('<')) {
+      return cloneElement(elem, attrs, value);
+    }
 
-    // The formatted parts can be passed to `cloneElements` as arguments.  They
-    // will be used as children of the cloned element.
-    return cloneElement(elem, attrs, ...parts);
+    const translationNodes = Array.from(parseMarkup(value).childNodes);
+    const translatedChildren = translationNodes.map(childNode => {
+      if (childNode.nodeType === childNode.TEXT_NODE) {
+        return childNode.textContent;
+      }
+
+      // If the child is not expected just take its textContent.
+      if (!elems.hasOwnProperty(childNode.localName)) {
+        return childNode.textContent;
+      }
+
+      return cloneElement(
+        elems[childNode.localName],
+        // XXX Explcitly ignore any attributes defined in the translation.
+        null,
+        // Void elements (like input) must not have any children. If
+        // childNode.textContent is an empty string we need to pass null
+        // instead.
```

Comment:
I struggle to understand the comment. 
Why are you testing textContent in the context of void elements, and if it's an empty string you're returning null instead?
May be an indication that other readers will struggle with it as well.